src/controllers/action-queue/types/pushPayloadTypes.js

```diff
@@ -0,0 +1,33 @@
+// @flow
+
+import { type FirebaseMessagingTypes } from '@react-native-firebase/messaging'
+import { asObject, asString, asValue } from 'cleaners'
+import URL from 'url-parse'
+
+//
+// Payloads to be added to the PushMessages
+//
+
+export type PriceChangePayload = {
+  type: 'price-change',
+  pluginId: string,
+  body: string
+}
+
+const asPriceChangePayload = (message: FirebaseMessagingTypes.RemoteMessage): PriceChangePayload => {
+  const clean = asObject({
+    notification: asObject({
+      body: asString
+    }),
+    data: asObject({
+      type: asValue('price-change'),
+      pluginId: asString
+    })
+  })(message)
+  return { ...clean.notification, ...clean.data }
+}
+
+export const asPushMessagePayloadToEdgeUri = (data: FirebaseMessagingTypes.RemoteMessage): string => {
+  const { type, ...params } = asPriceChangePayload(data) // extend later with asEither
+  return `edge://${type}?${URL.qs.stringify(params)}`
+}
```

Comment:
These are not cleaners, since they take typed input (`FirebaseMessagingTypes.RemoteMessage`). So, their names should not start with `as`.

You should still make an `asPriceChange` cleaner with the `type: asValue('price-change'), pluginId: asString` bit, but then simply use it in the `pushMessagePayloadToEdgeUri` utility function. We don't need to clean the notification body stuff, since that's already type-safe.